src/main/java/com/rcv/ContestConfig.java

```diff
@@ -59,6 +58,65 @@
   // whether or not there are any validation errors
   private boolean isValid;
 
+  // function: loadContestConfig
+  // purpose: factory method to create ContestConfig from configPath
+  // - create rawContestConfig from file - can fail for IO issues or invalid json
+  // - validate rawContestConfig - can fail if certain elements do not exist
+  // - if the above succeed create and return ContestConfig wrapping rawContestConfig (cannot fail)
+  // returns: new ContestConfig object if checks pass otherwise null
+  static ContestConfig loadContestConfig(String configPath) {
+    if (configPath == null) {
+      Logger.log(Level.SEVERE, "No config path specified!");
+      return null;
+    }
+    // config will hold the new ContestConfig if construction succeeds
+    ContestConfig config = null;
+    // set config file parent folder as default user folder
+    FileUtils.setUserDirectory(new File(configPath).getParent());
+
+    // rawConfig holds the basic contest config data parsed from json
+    // this will be null if there is a problem loading it
+    RawContestConfig rawConfig = JsonParser.readFromFile(configPath, RawContestConfig.class);
+    if (rawConfig == null) {
+      Logger.log(Level.SEVERE, "Failed to load contest config: %s", configPath);
+    } else {
+      Logger.log(Level.INFO, "Loaded: %s", configPath);
+      // perform some additional sanity checks
+      if (validateRawConfigFields(rawConfig)) {
+        // checks passed so create the ContestConfig
+        config = new ContestConfig(rawConfig);
+      } else {
+        Logger.log(Level.SEVERE, "Failed to create contest config!");
+      }
+    }
+    return config;
+  }
+
+  // function: validateRawConfigFields
+  // purpose: perform some validation checks
+  // param: RawContestConfig to perform checks against
+  // returns: true if checks pass otherwise false
+  private static boolean validateRawConfigFields(RawContestConfig rawConfig) {
```

Comment:
It makes more sense to me to have this live inside `RawContestConfig`, since we've already got a `validate` function living here in `ContestConfig`.